### PR TITLE
(feat): Basemap selector [HE-813]

### PR DESCRIPTION
**Pull request by anamontiaga**
### Description
Add selector to integrate new basemap.

### Designs
[Figma Designs](https://www.figma.com/proto/2YAp3AS5lYCj2IWEAN8AJw/HE-Map-%3A-2022-2023?type=design&node-id=5065-53207&t=vx3cAweDCU6S1BjU-0&scaling=min-zoom&page-id=5065%3A49424&starting-point-node-id=5065%3A49443)

### Feature relevant tickets
[HE-813](https://vizzuality.atlassian.net/browse/HE-813)

[HE-813]: https://vizzuality.atlassian.net/browse/HE-813?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/containers/menus/basemap-selector/styles.module.scss [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    font-family: $font-family-1;
+    font-size: 8px;
+    font-weight: 600;
+    margin: 1px auto;
+    text-transform: uppercase;
+  }
+}
+
+.basemapDefault {
+  background-image: url('../../../assets/images/mini-default-basemap.png');
```
Comment (Bluesmile82) [inline src/containers/menus/basemap-selector/styles.module.scss:32]: is it possible to use a better url?

File: src/containers/menus/basemap-selector/styles.module.scss [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  transition: 0.3s;
+  &:hover {
+    border: 2px solid $brand-color-main-hover;
+  }
+
+
+  .basemapLabel {
+    color: $white;
+    font-family: $font-family-1;
+    font-size: 8px;
```
Comment (Bluesmile82) [inline src/containers/menus/basemap-selector/styles.module.scss:24]: Do we have something in typography for this to extend?

File: src/containers/menus/basemap-selector/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    font-weight: 600;
```
Comment (Bluesmile82) [inline src/containers/menus/basemap-selector/styles.module.scss:25]: Not sure if we have this font weight. Anyway, use the variables we have

File: src/containers/scenes/aoi-scene/index.js [hunk 4]
```diff
@@ -223,7 +223,8 @@ function AOIScene(props) {
     setBasemap({
       map,
       surfaceColor: '#070710',
```
Comment (Bluesmile82) [inline src/containers/scenes/aoi-scene/index.js:225]: we should use the surfaceColor in the config if we have it

File: src/utils/layer-manager-utils.js [hunk 5]
```diff
@@ -3,12 +3,13 @@ import { loadModules } from 'esri-loader';
 import intersection from 'lodash/intersection';
 
 import { LEGEND_FREE_LAYERS } from 'constants/layers-groups';
+// import { LANDCOVER_BASEMAP_LAYER } from 'constants/layers-slugs';
```
Comment (Bluesmile82) [inline src/utils/layer-manager-utils.js:6]: Please remove if the code is not used anymore

File: src/constants/layers-urls.js [hunk 6]
```diff
@@ -151,6 +152,8 @@ const GADM_0_ADMIN_AREAS_FEATURE_LAYER_URL =
 const GADM_1_ADMIN_AREAS_FEATURE_LAYER_URL =
   'https://utility.arcgis.com/usrsvcs/servers/8f42e91efa7b43d1a38afa70dac9684f/rest/services/gadm1_precalculated_aoi_summaries_SPI/FeatureServer';
 export const LAYERS_URLS = {
+  [LANDCOVER_BASEMAP_LAYER]:
```
Comment (Bluesmile82) [inline src/constants/layers-urls.js:155]: We dont need this anymore, we have the id. Please remove it

File: src/constants/layers-slugs.js [hunk 7]
```diff
@@ -1,3 +1,6 @@
+// Basemaps
+export const LANDCOVER_BASEMAP_LAYER = 'landcover_basemap';
```
Comment (Bluesmile82) [inline src/constants/layers-slugs.js:2]: We dont need this

File: src/containers/scenes/aoi-scene/index.js [hunk 8]
```diff
@@ -223,7 +223,8 @@ function AOIScene(props) {
     setBasemap({
```
Comment (Bluesmile82) [inline src/containers/scenes/aoi-scene/index.js:223]: we should set the basemap only once on the map initialization and then use the setBasemap only on the scene component

File: src/utils/Sentinel2_LULC_basemap.json
Comment (Bluesmile82) [inline src/utils/Sentinel2_LULC_basemap.json:1]: Lets remove this too

File: src/utils/layer-manager-utils.js (hunk 5 above)
Comment (Bluesmile82) [inline src/utils/layer-manager-utils.js:6]: Remove this if not used

File: src/containers/menus/basemap-selector/styles.module.scss (hunk 2 above)
Comment (anamontiaga) [reply src/containers/menus/basemap-selector/styles.module.scss:24]: We dont have one with this fontsize

Comment (Bluesmile82) [reply src/containers/menus/basemap-selector/styles.module.scss:24]: Maybe you can add one then. So if we have this font-size again we will have one

Comment (anamontiaga) [reply src/containers/menus/basemap-selector/styles.module.scss:24]: Great

**Comment by Bluesmile82** [comment]
Thanks for the changes!

**Pull request merged**